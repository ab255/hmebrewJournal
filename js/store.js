const store = [
  {beerName: 'Green Bomber', beerType: 'Imperial IPA', brewDate: '4/28/2013'},
  {beerName: 'Red Dawn', beerType: 'American Amber Ale', brewDate: '3/26/2013'}
];

module.exports = store;
