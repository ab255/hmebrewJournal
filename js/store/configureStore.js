import { AsyncStorage } from 'react-native';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { persistStore, autoRehydrate } from 'redux-persist';

import reducers from '../reducers/reducers';


const createBrewStore = applyMiddleware(thunk)(createStore);

const configureStore = () => {
  const store = autoRehydrate()(createBrewStore)(reducers);
  persistStore(store, {storage: AsyncStorage});
  return store;
}

export default configureStore;
