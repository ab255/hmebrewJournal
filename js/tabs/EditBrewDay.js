import React, {Component} from 'react';
import {
  DatePickerIOS,
  ScrollView,
  StyleSheet,
  Text,
  TextInput,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import moment from 'moment';

class EditBrewDay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      BrewDate: this.props.brewDate,
      TimeStarted: this.props.timeStarted,
      Weather: this.props.weather,
      MashNotes: this.props.mashNotes,
      BoilNotes: this.props.boilNotes,
      PostBoilNotes: this.props.postBoilNotes,
      OgReading: this.props.ogReading,
      datePickerMode: 'hidden',
    }
  }

  toggleDatePicker = () => {
    let mode = this.state.datePickerMode =='hidden' ? 'visible' : 'hidden';
    this.setState({ datePickerMode: mode });
  }

  onDateChange = (date) => {
    this.setState({ BrewDate: date })
  }

  render() {
    let datePicker = (
      <View style={styles.datePicker}>
        <DatePickerIOS
          date={new Date(this.state.BrewDate)}
          mode='date'
          onDateChange={this.onDateChange.bind(this)}
          stlye={styles.datePicker}
        />
      </View>
    )
    return (
      <KeyboardAwareScrollView style={styles.container}>
        <View>
          <View>
          <Text style={styles.title}>Brew Date:</Text>
            <TouchableWithoutFeedback onPress={ this.toggleDatePicker.bind(this) }>
              <View style={styles.dateField}>
                <Text style={styles.dateText}>
                  {moment(this.state.BrewDate).format('MMMM Do YYYY')}
                </Text>
              </View>
            </TouchableWithoutFeedback>
          </View>
          { this.state.datePickerMode == 'visible' ? datePicker : <View/> }
          <Text style={styles.title}>(Click Date To Toggle Date Picker)</Text>
        </View>
        <View>
          <Text style={styles.title}>Start Time:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(TimeStarted) => this.setState({TimeStarted})}
            value={this.state.TimeStarted}
            keyboardType='numbers-and-punctuation'
            placeholder='Start of brewing time'
          />
        </View>
        <View>
          <Text style={styles.title}>Current Weather:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(Weather) => this.setState({Weather})}
            value={this.state.Weather}
            keyboardType='decimal-pad'
            placeholder='Current temperature'
          />
        </View>
        <View>
          <Text style={styles.title}>Mash/Steep Notes:</Text>
          <TextInput
            multiline={true}
            style={styles.ingredientsInput}
            onChangeText={(MashNotes) => this.setState({MashNotes})}
            value={this.state.MashNotes}
            placeholder='Mash or steep notes (if any)'
          />
        </View>
        <View>
          <Text style={styles.title}>Boil Notes:</Text>
          <TextInput
            multiline={true}
            style={styles.ingredientsInput}
            onChangeText={(BoilNotes) => this.setState({BoilNotes})}
            value={this.state.BoilNotes}
            placeholder='Add boil notes'
          />
        </View>
        <View>
          <Text style={styles.title}>Post Boil Notes:</Text>
          <TextInput
            multiline={true}
            style={styles.ingredientsInput}
            onChangeText={(PostBoilNotes) => this.setState({PostBoilNotes})}
            value={this.state.PostBoilNotes}
            placeholder='Add post boil notes'
          />
        </View>
        <View>
          <Text style={styles.title}>Original Gravity:</Text>
          <TextInput
            style={styles.brewInput}
            onChangeText={(OgReading) => this.setState({OgReading})}
            value={this.state.OgReading}
            placeholder='Original gravity ex. 1.065'
            keyboardType='decimal-pad'
          />
        </View>
        <TouchableHighlight
        >
          <View style={styles.submitButton}>
            <Text style={styles.buttonText}>Submit</Text>
          </View>
        </TouchableHighlight>
      </KeyboardAwareScrollView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#911F27',
    flex: 1,
    padding: 10,
  },
  dateField: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    height: 30,
    justifyContent: 'center',
    padding: 5,
    marginBottom: 5,
  },
  dateText: {
    color: '#630A10',
    fontSize: 16,
  },
  datePicker: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    marginBottom: 5,
  },
  title: {
    color: '#F7D098',
    fontWeight: 'bold',
    marginBottom: 5,
  },
  brewInput: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    color: '#630A10',
    height: 30,
    paddingLeft: 5,
    marginBottom: 10,
  },
  ingredientsInput: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    color: '#630A10',
    paddingLeft: 5,
    height: 100,
    fontSize: 16,
    marginBottom: 10,
  },
  submitButton: {
    alignSelf: 'center',
    backgroundColor: '#F7D098',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 5,
    width: 300,
    height: 30,
    margin: 20,
    },
  buttonText: {
    color: '#630A10',
    fontSize: 16,
  },
})

export default EditBrewDay;
