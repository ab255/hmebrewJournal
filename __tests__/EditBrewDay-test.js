import 'react-native';
import React from 'react';
import EditBrewDay from '../js/tabs/EditBrewDay.js';
import store from '../js/mockStore.js'

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  let beer = store
  const tree = renderer.create(
    <EditBrewDay route={ { beer } } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
