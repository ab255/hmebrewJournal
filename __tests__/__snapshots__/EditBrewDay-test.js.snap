exports[`test renders correctly 1`] = `
<ScrollView
  contentInset={
    Object {
      "bottom": 49,
    }
  }
  enableAutoAutomaticScroll={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  onScroll={[Function]}
  scrollEventThrottle={8}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#911F27",
      "flex": 1,
      "padding": 10,
    }
  }>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#F7D098",
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }>
        Brew Date:
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFF",
            "borderRadius": 5,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 5,
            "padding": 5,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#630A10",
              "fontSize": 16,
            }
          }>
          April 28th 2013
        </Text>
      </View>
    </View>
    <View />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      (Click Date To Toggle Date Picker)
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Start Time:
    </Text>
    <TextInput
      keyboardType="numbers-and-punctuation"
      onChangeText={[Function]}
      placeholder="Start of brewing time"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Current Weather:
    </Text>
    <TextInput
      keyboardType="decimal-pad"
      onChangeText={[Function]}
      placeholder="Current temperature"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="70" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Mash/Steep Notes:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Mash or steep notes (if any)"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "fontSize": 16,
          "height": 100,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Boil Notes:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Add boil notes"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "fontSize": 16,
          "height": 100,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Post Boil Notes:
    </Text>
    <TextInput
      multiline={true}
      onChangeText={[Function]}
      placeholder="Add post boil notes"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "fontSize": 16,
          "height": 100,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="" />
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#F7D098",
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }>
      Original Gravity:
    </Text>
    <TextInput
      keyboardType="decimal-pad"
      onChangeText={[Function]}
      placeholder="Original gravity ex. 1.065"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "color": "#630A10",
          "height": 30,
          "marginBottom": 10,
          "paddingLeft": 5,
        }
      }
      value="1.085" />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F7D098",
          "borderRadius": 5,
          "height": 30,
          "justifyContent": "center",
          "margin": 20,
          "width": 300,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#630A10",
            "fontSize": 16,
          }
        }>
        Submit
      </Text>
    </View>
  </View>
</ScrollView>
`;
